### ISSUE TITLE
Cypress/react CT 5.1.1 does not resolve modules using tsConfig.js baseUrl

### ISSUE THREAD

**Issue by yann-combarnous** [cypress-io/cypress#15514]
### Current behavior
I tried to upgrade from cypress/react 4.16.7 to 5.1.1 and encountered 2 issues (using CRA 4.0.3 and Cypress 6.7.1):
1- first failure was due to missing webpack-dev-server installation (not mentioned in the documentation)
2- second failure is due to module resolution failing when trying to run CT test:
```
Error: Webpack Compilation Error
./src/api/restCallApi.ts
Module not found: Error: Can't resolve 'mock_api' in '/Users/xyz/Code/portal/src/api'
resolve 'mock_api' in '/Users/xyz/Code/portal/src/api'
  Parsed request is a module
  using description file: /Users/xyz/Code/portal/package.json (relative path: ./src/api)
    Field 'browser' doesn't contain a valid alias configuration
    Looked for and couldn't find the file at the following paths:
[/Users/xyz/Code/portal/src/api/node_modules]
 ```

Our root TSConfig mentions baseUrl for relative path resolution:
- baseUrl: 'src'

### Desired behavior
Relative path are supported and honoring TSconfig.

Module resolution should have found 'mock_api' in 'src/mock_api'.

### Versions
Cypress 6.7.1
CRA 4.0.3
Typescript 4.2.3
@Cypress/react 5.1.1
@cypress/webpack-dev-server 1.1.0

**Comment by dmtrKovalenko** [comment]
This likely happens because your webpack is not handling tsconfig base url. If you are using babel or using typescript only for type checking it won't affect how modules are resolved

**Comment by yann-combarnous** [comment]
@dmtrKovalenko, we use off-the-shelf create-react-app, so no webpack or babel changes, we use defaults.

This is the description of the setup we used: https://medium.com/@mikecripps/absolute-imports-with-create-react-app-d70fb65ea012.

So it looks like this is an issue with your CT webpack-dev-server resolution, don't you think?

**Comment by lmiller1990** [comment]
Hi @yann-combarnous,

Thanks for the patience and reply. It could be an issue on our end. 

We just released Cypress 7.0, which features a dedicated runner for component testing (launch via `yarn cypress open-ct`). Can you upgrade and see if it's still an issue?

Component Testing intro: https://docs.cypress.io/guides/component-testing/introduction
Changelog: https://docs.cypress.io/guides/references/changelog
Migration guide: https://docs.cypress.io/guides/references/migration-guide#Component-Testing

If you are still having the problem, can you post a reproduction repository so I can debug it? Thanks.

**Comment by yann-combarnous** [comment]
@lmiller1990 , thx, will try and upgrade as soon as the performance and intercept issues are resolved in 7.0.

**Comment by lmiller1990** [comment]
Sure, let's wait and see. I'll keep this open until you can confirm the fix has worked.
